### PR TITLE
UAR-1254 Set page links on first Remove screen

### PR DESCRIPTION
**Pull request by mwestacott**
### JIRA link

https://companieshouse.atlassian.net/browse/UAR-1254

### Change description

* Back link on first page now redirects to remove landing page
* Remove landing page is configurable and value read from env config
* Feedback link different to Update journey
* Remove title correctly set after authentication redirect
* Unit tests added

### Work checklist

- [X] Tests added where applicable
- [ ] UI changes meet accessibility criteria

### Merge instructions

We are committed to keeping commit history clean, consistent and linear. To achieve this, this commit should be structured as follows:

```
<type>[optional scope]: <description>
```

and contain the following structural elements:

- fix: a commit that patches a bug in your codebase (this correlates with PATCH in semantic versioning),
- feat: a commit that introduces a new feature to the codebase (this correlates with MINOR in semantic versioning),
- BREAKING CHANGE: a commit that has a footer `BREAKING CHANGE:` introduces a breaking API change (correlating with MAJOR in semantic versioning). A BREAKING CHANGE can be part of commits of any type,
- types other than `fix:` and `feat:` are allowed, for example `build:`, `chore:`, `ci:`, `docs:`, `style:`, `refactor:`, `perf:`, `test:`, and others,
- footers other than `BREAKING CHANGE: <description>` may be provided.

### CODE REVIEW

File: src/middleware/authentication.middleware.ts
Comment (mattch1) [inline src/middleware/authentication.middleware.ts:57]: So the presence of the parameter acts as a flag

File: terraform/groups/ecs-service/profiles/staging-eu-west-2/staging/vars
Comment (mattch1) [inline terraform/groups/ecs-service/profiles/staging-eu-west-2/staging/vars:21]: Will it be the flag or the url used when deciding whether to an update page or a remove page? e.g. the fork from the "Who can we contact about this filing page?"

Comment (mwestacott) [reply terraform/groups/ecs-service/profiles/staging-eu-west-2/staging/vars:21]: It's the URL parameter, which is ultimately part of the URL. Currently, if not present then it's the update journey, if it is present AND its value is 'remove' then it's the remove journey.

File: src/middleware/authentication.middleware.ts
Comment (mattch1) [reply src/middleware/authentication.middleware.ts:57]: So does the use of a value here rather than a flag or effective flag mean that we will introduce a query parameter for the update journey as well?

File: terraform/groups/ecs-service/profiles/staging-eu-west-2/staging/vars
Comment (mattch1) [reply terraform/groups/ecs-service/profiles/staging-eu-west-2/staging/vars:21]: So we keep open the possibility of introducing a parameter value for update by choosing to use a value but will not do now, which means this functions like a flag base don presence or absance.

File: src/middleware/authentication.middleware.ts
Comment (mwestacott) [reply src/middleware/authentication.middleware.ts:57]: Not necessarily, but we could use the `journey` query parameter to indicate an update - or even register - journey, if we need to. All three journey types are defined in the `JourneyType` enum.

File: terraform/groups/ecs-service/profiles/staging-eu-west-2/staging/vars
Comment (mwestacott) [reply terraform/groups/ecs-service/profiles/staging-eu-west-2/staging/vars:21]: Nearly. So absence of the parameter essentially indicates the update journey (for now, as no code is checking this) but presence (alone) is not enough - the value must be `remove` for the app to recognise it as the remove journey.

**Review by mattch1 — APPROVED** [review]
LGTM - pending other approvals.

File: README.md
Comment (markpit) [inline README.md:89]: i think this is meant to be the url to start the journey in test envs, so perhaps the 'choose resume / new " page?

File: src/middleware/authentication.middleware.ts
Comment (markpit) [inline src/middleware/authentication.middleware.ts:55]: i would add into the comments 
Ensure that user is placed on the Remove (not Update) journey after logging in **by re-adding the journey=remove url query param**

File: src/middleware/authentication.middleware.ts
Comment (markpit) [inline src/middleware/authentication.middleware.ts:60]: does the else block need changing also on line 60, I think some (all?) remove screens will start with  UPDATE_LANDING_URL

File: terraform/groups/ecs-service/profiles/development-eu-west-2/cidev/vars
Comment (markpit) [inline terraform/groups/ecs-service/profiles/development-eu-west-2/cidev/vars:18]: i thought the landing page would be /update-an-overseas-entity/continue-with-saved-filing?journey=remove and that /gov-uk-remove-guidance-page would be the 'back' link from the landing page

File: src/middleware/authentication.middleware.ts
Comment (mwestacott) [reply src/middleware/authentication.middleware.ts:60]: I think we agreed to leave this where it is (at least for now).

File: README.md
Comment (mwestacott) [reply README.md:89]: Changed this and also added some start URLs in the 'Running locally on Docker env' section above.

File: terraform/groups/ecs-service/profiles/development-eu-west-2/cidev/vars
Comment (mwestacott) [reply terraform/groups/ecs-service/profiles/development-eu-west-2/cidev/vars:18]: I think we clarified this in discussion yesterday - I've now made these static HTTP URLs that don't 'point into' our web app journey. So a bit more like the config in LIVE.

File: README.md
Comment (dlloyd1-ch) [inline README.md:89]: Do we also need to add an endpoint for remove in the Endpoints section (line 72)?

**Review by markpit — APPROVED** [review]

**Review by dlloyd1-ch — COMMENTED** [review]
Looks good. One minor comment.

File: README.md
Comment (mwestacott) [reply README.md:89]: Yes, we should. Will add one and push change to [this](https://github.com/companieshouse/overseas-entities-web/pull/1200) PR.

**Pull request merged**
